assets/html/js/footnote.js

```diff
@@ -0,0 +1,10 @@
+// libraries: jquery
+// arguments: $
+
+$(document).ready(function() {
```

Comment:
I think this is where it actually sets the preview text that will be shown on hover, right?

Can you elaborate a little on what assumptions this makes on the HTML of the footnote? I'm trying to figure out if these assumptions hold / could be made to hold for bibliographic references as well.

Reply:
Actually, this is used to get the footnote content in hidden span for preview! This jQuery code takes the footnote definitions from the end of the page and puts it into the hidden spans just under the footnotes based on their footnote IDs!

For bibliographic references, we will need another way because the references are in another file and not on same page!
We can add a hidden span in Julia part like this:
```
span[".footnote-preview", :id => "fn-$(f.id)"](footnote content comes here using that JQuery!)
```
We can add a small julia function which stores bibliographic reference on pages its being used on and then add them in span! This way we can use this method for all things that require hover preview feature whether it is footnotes, bibliographic references or docstring references!

We will just need those small julia functions for storing content and put them in span and similar css can be used for all, No Javascript or JQuery will be needed and this is most efficient way!

I used JQuery because I was not able to fetch content using Julia (lack of my expertise)